internal/policy/targets.go

```diff
@@ -96,6 +96,27 @@ func UpdateDelegation(targetsMetadata *tuf.TargetsMetadata, ruleName string, aut
 	return targetsMetadata, nil
 }
 
+// ReorderDelegations changes the order of delegations in TargetsMetadata.
+func ReorderDelegations(targetsMetadata *tuf.TargetsMetadata, ruleNames []string) (*tuf.TargetsMetadata, error) {
+	allDelegations := []tuf.Delegation{}
+	for _, ruleName := range ruleNames {
+		if ruleName == AllowRuleName {
+			return nil, ErrCannotManipulateAllowRule
+		}
+
+		for _, delegation := range targetsMetadata.Delegations.Roles {
+			if delegation.Name == ruleName {
+				allDelegations = append(allDelegations, delegation)
+			}
+		}
```

Comment:
Instead of doing this for every single rulename, can we first precompute this in a map of rulename to roles? Then, when we iterate through the rule names, we can query the roles for that rule name and append it to allDelegations instead of iterating multiple times.

Reply:
Thanks, I'll try that with a pre-computed map